.github/workflows/package-json-protection.yaml

```diff
@@ -0,0 +1,31 @@
+name: package.json Protection
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  package-json-protection:
+    name: package.json Protection
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        # Required since "HEAD^" is passed to Get-ChangedFiles
+        fetch-depth: 2
+
+    - name: Detect changes to package[-lock].json
+      run: |
+        . eng/scripts/ChangedFiles-Functions.ps1
+        $changedFiles = @(Get-ChangedFiles -baseCommitish HEAD^ -targetCommitish HEAD -diffFilter "")
+        if (($changedFiles -contains "package.json") -or ($changedFiles -contains "package-lock.json")) {
+          Write-Output "::error::'package.json' and 'package-lock.json' should only be updated by the Azure SDK team.  If the changes are intentional, the PR may be merged by the Azure SDK team via bypassing the branch protections."
```

Comment:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-a-notice-message

Should we create an annotation on the file so it shows up in the code review flow?

Reply:
Should be working now.  See the test PR to verify: https://github.com/Azure/azure-rest-api-specs/pull/30350